src/components/character-count/character-count.ts

```diff
@@ -0,0 +1,39 @@
+import Vue, { PluginObject } from 'vue';
+import Component from 'vue-class-component';
+import { Prop } from 'vue-property-decorator';
+
+import { CHARACTER_COUNT_NAME } from '../component-names';
+import WithRender from './character-count.html?style=./character-count.scss';
+
+@WithRender
+@Component
+export class MCharacterCount extends Vue {
+    @Prop()
+    public valueLength: number;
+    @Prop()
+    public maxLength?: number;
+    @Prop({ default: 0 })
+    public threshold: number;
+    @Prop({ default: true })
+    public transition: boolean;
+
+    private get hasCounter(): boolean {
+        return this.maxLength && this.maxLength > 0 ? this.isThresholdValid : false;
```

Comment:
Si `this.maxLength` est `undefined`, `this.maxLength > 0` va retourner systématiquement `false`. Donc cette ligne pourrait se résumer à

`return this.maxLength > 0 && this.isThresholdValid`